### PR TITLE
Create separate java_tools for darwin_x86_64 and darwin_arm64

### PR DESCRIPTION
**Pull request by hvadehra**
This is step 2 of https://github.com/bazelbuild/bazel/issues/17780

### CODE REVIEW

**Review by comius — APPROVED** [review]

File: src/BUILD [hunk 1]
```diff
@@ -537,7 +537,8 @@ sh_binary(
         "tmp/build",
         "--platform",
     ] + select({
-        "//src/conditions:darwin": ["darwin"],
```
Comment (keith) [inline src/BUILD:540]: Hrm yea still not sure if you need Darwin here, I think you do even if it points to the x86 version still

**Review by keith — APPROVED** [review]

File: src/BUILD (hunk 1 above)
Comment (hvadehra) [reply src/BUILD:540]: Could you elaborate on why we might want to contine selecting more generally on "darwin"?

AFAIK this script is only invoked in a buildkite pipeline that (after this change) will be run on both darwin_x86_64 and darwin_arm64.

Comment (keith) [reply src/BUILD:540]: so after https://github.com/bazelbuild/bazel/pull/17547 this won't be possible, but today my understanding is that on a x86_64 mac, if you don't pass any CPU and you're using the CLT toolchain, you will get `darwin` as a CPU. If you explicitly pass `--cpu=darwin_x86_64` then you won't ever hit it, but that weirdness is the goal of my other PR. but we have to merge this, then re-merge my toolchain PR, and then the darwin CPU PR before that is true

Comment (hvadehra) [reply src/BUILD:540]: I see, I don't think that should be an issue in this case. We won't be using any flags or the CLT toolchain.

**Pull request closed without merging**